### ISSUE TITLE
Build fails - can't find ld.lld-5.0

### ISSUE THREAD

**Issue by thejpster** [embed-rs/stm32f7-discovery#26]
I get this error when building with xargo. Is there a dependency missing from the documentation, or have I got something misconfigured?

I'm running Fedora 25.

```
$ xargo build
   Compiling stm32f7_discovery v0.1.0 (file:///home/jonathan/Documents/programming/stm32f7-discovery)
error: could not exec the linker `ld.lld-5.0`: No such file or directory (os error 2)
  |
  = note: "ld.lld-5.0" "-L" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/stm32f7_discovery-8bd3543bbcf36c0f.0.o" "-o" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/stm32f7_discovery-8bd3543bbcf36c0f" "--gc-sections" "-L" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps" "-L" "/home/jonathan/Documents/programming/stm32f7-discovery/target/debug/deps" "-L" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib" "-Bstatic" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libr0-4c9ac95f64aec4f1.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libstm32f7_discovery-6153ba08cda4cfc2.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libspin-1433c0b45b26eec6.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libembedded_stm32f7-51243b9927c774bd.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libonce-ca6431046f69cecf.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libembedded-0b061f7c242587d9.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libvolatile-e95a9c93c95b4d05.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libbit_field-a45edf4fa6f9a542.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libbitflags-b857f06d8702b68e.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libarrayvec-45a597b5892cf267.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libnodrop-e6d7a15cb157c190.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libodds-4d86427aaea60869.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libbit_field-60b9c6c0d7fc80d3.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libbit_field-9493560445ae6fc0.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libnet-70d5d8d662ef3204.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libbyteorder-574f80dfed4c7759.rlib" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib/libcollections-3258484305ade22a.rlib" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib/liballoc-b141cff101af6ec4.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/liballoc_cortex_m-4b9118521c562789.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/liblinked_list_allocator-4596ac83bd7de26f.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libcortex_m-e3e3b27db187a72c.rlib" "/home/jonathan/Documents/programming/stm32f7-discovery/target/stm32f7/debug/deps/libvolatile_register-bd1a05ad8326c8e1.rlib" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib/libstd_unicode-9c985ead028450d3.rlib" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib/libcore-38427e0ea84eda51.rlib" "/home/jonathan/.xargo/lib/rustlib/stm32f7/lib/libcompiler_builtins-48645c160d69be81.rlib" "-Bdynamic"

error: aborting due to previous error

error: Could not compile `stm32f7_discovery`.

To learn more, run the command again with --verbose.
```

**Comment by oli-obk** [comment]
@phil-opp  did you get lld working? If not we should probably revert it for now.

@thejpster you can change the linker back to GCC by editing the stm32f7.json. replace the flavour value wit gcc  and the linker name wit ld

**Comment by phil-opp** [comment]
No, I couldn't get it working. Reverting it is probably the best option at the moment

**Comment by thejpster** [comment]
https://github.com/embed-rs/stm32f7-discovery/pull/27

**Comment by f-bro** [comment]
Is there a way to switch to GCC on travis? I think, this was the reason why @phil-opp switched to lld.

We could try:
```
addons:
  apt:
    sources:
      - debian-sid
    packages:
      - gcc-arm-none-eabi
```
https://github.com/travis-ci/apt-package-whitelist/pull/34

**Comment by thejpster** [comment]
Travis works for me on [stellaris-launchpad](https://github.com/thejpster/stellaris-launchpad). I use a ppa to get GCC.

**Comment by f-bro** [comment]
Yes, but I think @phil-opp didn't want to use sudo to speed up the build process. But I'm not sure. Let's wait for @phil-opp opinion.

**Comment by oli-obk** [comment]
A working Travis is better than a fast but broken Travis. Let's do

**Comment by phil-opp** [comment]
Fixed in #28

> A working Travis is better than a fast but broken Travis. Let's do

Agreed. Shall I do it or is someone working on it already?

**Comment by f-bro** [comment]
Check #29
